src/fs/file_format.c

```diff
@@ -14,6 +14,10 @@
 enum fformat raw_get_file_format(uint8_t *file) {
 	/* signature of PNG files */
 	uint8_t png_signature[8] = {0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a};
+	/* signature of ELF files */
+	uint8_t elf_signature[4] = {0x7f, 0x45, 0x4c, 0x46};
+	/* signature of XML files */
+	uint8_t xml_signature[21] = {0x3c, 0x3f, 0x78, 0x6d, 0x6c, 0x20, 0x76, 0x65, 0x72, 0x73, 0x69, 0x6f, 0x6e, 0x3d, 0x22, 0x31, 0x2e, 0x30, 0x22, 0x3f, 0x3e};
```

Comment:
As I can see, you check for XML version 1.0 here. Why limit it to single verison?

It would be nice to print XML version in file command (but that's not necessary)

P.S. This line is way too long, it's recommended to stick to 80 symbols per line.

Reply:
Should I just check for "<?xml" in the signature?

Reply:
Yes, it would be better.